src/lib.rs

```diff
@@ -1310,9 +1310,13 @@ pub trait Itertools : Iterator {
     /// assert!(data.into_iter().all_equal());
     /// ```
     fn all_equal(&mut self) -> bool
-        where Self::Item: PartialEq,
+        where Self: Sized,
```

Comment:
I think requiring `Sized` is ok. Can anyone with more expertise confirm this is unproblematic?

Reply:
There are already lots of methods you can't call on an unsized iterator, so I don't think this is problematic.

Reply:
I believe the documentation (sometimes?) leaves out `Self: Sized` bounds because they’re so common 😕

Reply:
I believe the documentation (sometimes?) leaves out `Self: Sized` bounds because they’re so common 😕

Reply:
Per [Reddit](https://www.reddit.com/r/rust/comments/cgh56g/hey_rustaceans_got_an_easy_question_ask_here/eukw73y/?context=1&depth=3), I think this is a bug. I've filed an issue: https://github.com/rust-lang/rust/issues/62899.